pkg/daemon/update.go

```diff
@@ -406,6 +464,22 @@ func (dn *Daemon) writeFiles(files []ignv2_2types.File) error {
 	return nil
 }
 
+// seperated out writeSSHKeys from updatePasswd even though this is the only update allowed to Passwd right now
+func (dn *Daemon) writeSSHKeys(oldUsers []ignv2_2types.PasswdUser, newUsers []ignv2_2types.PasswdUser, userKeys []int) error {
+	var path string
+	for _, idx := range userKeys {
+		if !reflect.DeepEqual(oldUsers[idx].SSHAuthorizedKeys, newUsers[idx].SSHAuthorizedKeys) {
+			// ssh keys are in PasswdUser.HomeDir/.ssh
+			path = filepath.Join(oldUsers[idx].HomeDir, ".ssh")
+			glog.V(2).Infof("Writing SSHKeys at %q:", path)
+			// write the SSHKeys
+		} else {
+			return fmt.Errorf("Failed to write SSHKeys")
```

Comment:
Once the ssh key writing is added I recommend returning the error that is bubbled up from that work OR adding a log statement with details before this return